### PR TITLE
use Optional<T> instead of T? to workaround SR-11777

### PR DESCRIPTION
**Pull request by weissi**
Motivation:

In Swift, writing

    var something: T?

    init() {
        self.something = someValue
    }

means that the compiler will first set `self.something` to `nil` and
then in the init override it with `self.someValue`
(https://bugs.swift.org/browse/SR-11777). Unfortunately, because of
https://bugs.swift.org/browse/SR-11768 , stored property initialisation
cannot be made `@inlinable` (short of using `@frozen` which isn't
available in Swift 5.0).

The combination of SR-11768 and SR-11777 leads to `var something: T?`
having much worse code than `var something: Optional<T>` iff the `init`
is `public` and `@inlinable`.

Modifications:

Change all `var something: T?` to `var something: Optional<T>`

Result:

Faster code, sad NIO developers.

### CODE REVIEW

File: Sources/NIO/Bootstrap.swift [hunk 1]
```diff
@@ -354,11 +356,11 @@ private extension Channel {
 public final class ClientBootstrap {
 
     private let group: EventLoopGroup
-    private var channelInitializer: ((Channel) -> EventLoopFuture<Void>)?
+    private var channelInitializer: Optional<(Channel) -> EventLoopFuture<Void>>
```
Comment (Davidde94) [inline Sources/NIO/Bootstrap.swift]: Nit, maybe out of scope, now that this line is a little more "complex" a typealias might be nice

File: Sources/NIO/EventLoop.swift [hunk 2]
```diff
@@ -58,14 +58,15 @@ public final class RepeatedTask {
     private let delay: TimeAmount
     private let eventLoop: EventLoop
     private let cancellationPromise: EventLoopPromise<Void>?
-    private var scheduled: Scheduled<EventLoopFuture<Void>>?
-    private var task: ((RepeatedTask) -> EventLoopFuture<Void>)?
+    private var scheduled: Optional<Scheduled<EventLoopFuture<Void>>>
+    private var task: Optional<(RepeatedTask) -> EventLoopFuture<Void>>
```
Comment (Davidde94) [inline Sources/NIO/EventLoop.swift]: Nit: See previous comment

File: Sources/NIO/Bootstrap.swift (hunk 1 above)
Comment (weissi) [reply Sources/NIO/Bootstrap.swift]: @Davidde94 hmm, what name would you propose for a typealias for `(Channel) -> EventLoopFuture<Void>`?

Comment (Davidde94) [reply Sources/NIO/Bootstrap.swift]: `ChannelInitializer`?

Comment (weissi) [reply Sources/NIO/Bootstrap.swift]: doesn't that make it harder to see what that is? CC @Lukasa I'm open to making it `Optional<ChannelInitializer>` but I think that's more opaque than `Optional<(Channel) -> ELF<Void>>` but maybe it's me?

Comment (Lukasa) [reply Sources/NIO/Bootstrap.swift]: I don't love `ChannelInitializer` as a typealias: `ChannelInitializerCallback` might be better.

Comment (Davidde94) [reply Sources/NIO/Bootstrap.swift]: Either sound good to me - note this was a readability nit, I think the good is good to go regardless :)

Comment (weissi) [reply Sources/NIO/Bootstrap.swift]: @Davidde94 / @Lukasa fixed.

File: Sources/NIO/EventLoop.swift (hunk 2 above)
Comment (weissi) [reply Sources/NIO/EventLoop.swift]: done too

**Pull request merged**